### ISSUE TITLE
App crash when use rotate and flip filter

### ISSUE THREAD

**Issue by tianxu0836** [burhanrashid52/PhotoEditor#182]
Tap the “Filter” option.

Select the “Rotate” filter.

Select the “Flip Vertical” filter.

Repeat step 4 and 5 until the app crashes.

**Comment by burhanrashid52** [comment]
Since the filter works asynchronously using GLSurfaceView and requires a lot of memory. Doing those very fast may lead to `OutOfMemoryException`. I am looking for an alternative solution for this. Renderscript might solve not sure 🤔 . Any input is appreciate.

**Comment by tianxu0836** [comment]
if I put a progress bar and block the UI after it is done, is it going to
avoid the problem?
The thing is how do I know the filter is done and memory released?

**Comment by burhanrashid52** [comment]
Yes. That's a mistake we did in the implementation. Need to fix that
